prog/dftb+/lib_dftb/dispmbd.F90

```diff
@@ -300,13 +300,13 @@ end function getRCutoff
 
 
   !> Update charges in the MBD model
-  subroutine updateOnsiteCharges(this, qOnsite, orb, referenceN0, species0, tCanUseCharges)
+  subroutine updateOnsiteCharges(this, qNetAtom, orb, referenceN0, species0, tCanUseCharges)
 
     !> Instance
     class(TDispMbd), intent(inout) :: this
 
     !> Net charges
-    real(dp), intent(in) :: qOnsite(:)
+    real(dp), intent(in), allocatable :: qNetAtom(:)
```

Comment:
Have gone with the assert, as the interface above it could reasonably be called with it the variable unallocated. 

What do you think about making the interface just updateCharges with arguments including qAtom and qNetAtom (or qOnsite as was)? Then dispersion models using the full charges can pick those up, MBD can get qNetAtom if allocated and models that aren't charge dependent can ignore all of it.

Reply:
We should probably follow the already existing interfaces like the third order or solvation one, they are general enough to allow most extensions without requiring to modify the dispersion interface again.